python/lsst/analysis/drp/calcFunctors.py

```diff
@@ -71,6 +75,39 @@ def __call__(self, df):
         return magDiff
 
 
+class ExtinctionCorrectedMagDiff(DataFrameAction):
+    magDiff = ConfigurableActionField(doc="Action that returns a difference in magnitudes",
+                                          default=MagDiff, dtype=DataFrameAction)
+    ebv = Field(doc="E(B-V) Column", dtype=str, default="ebv")
+    extinctionCoeffs = DictField(
+        doc="Dictionary of extinction coefficients for conversion from E(B-V) to extinction, A_filter."
+            "Key must be the band",
+        keytype=str, itemtype=float, optional=True,
+        default=None)
+
+    @property
+    def columns(self):
+        return self.magDiff.columns + (self.ebv,)
+
+    def __call__(self, df):
+        diff = self.magDiff(df)
+        if not self.extinctionCoeffs:
+            _LOG.warn("No extinction Coefficients. Not applying extinction correction")
+            return diff
+
+        ebv = df[self.ebv].values
+        col1Band = self.magDiff.col1.split('_')[0]
```

Comment:
Can we always guarantee that the band will be the first part of the split on `'_'` or does this need to be another configuration parameter?
(Also, I don't have an easy answer to this, but I was getting confused because I was reading `col` as `color` but here it means `column` and has been commented elsewhere we have some naming inconsistencies and technical debt, and is not going to be fixed on this PR.)

Reply:
Too late to change the naming convention of the Object Table now! 

But if you want the band to be optionally supplied per config parameters you got it.